src/node/blockstorage.cpp

```diff
@@ -299,6 +299,8 @@ void BlockManager::Unload()
     m_last_blockfile = 0;
     m_dirty_blockindex.clear();
     m_dirty_fileinfo.clear();
+
+    fHavePruned = false;
```

Comment:
In commit "Clear fHavePruned in BlockManager::Unload()" (18d68910bdd4f18eadbf72fa7ec0c0299def3666)

Note: `BlockManager::Unload()` is only called by `ChainstateManager::Unload()` and the `BlockManager` destructor. `ChainstateManager::Unload()` is only called by `UnloadBlockIndex`. So there is not really a change moving this line from `UnloadBlockIndex`